### DISCUSSION TITLE
Unable to compile the litttlenavmap and Cant follow deployment step

### DISCUSSION THREAD

**Discussion by Mohammad-Imran01** [albar965/littlenavmap#1066] [General]
This is the instructions i need to follow in the final steps but i cannot change the make command. I would like to know clearly about how to compile littlenavmap for further modification of the project.
1. How to call 'make deploy'
2. how to change the make command.
![Screenshot (58)](https://github.com/albar965/littlenavmap/assets/117526105/195208e3-bacf-4ba1-911e-473f23bc0a15)

This is the best i could follow till now.....

![Screenshot (59)](https://github.com/albar965/littlenavmap/assets/117526105/1bd0c258-2354-432f-bbc3-d6811a312a05)

**Comment by albar965** [comment]
What modifications do you plan? Would be good to know ahead to avoid double work.

In Qt Creator: Panel `Run` -> `Add Deploy Step` -> `Make`. Then enter `deploy` in 'Make arguments'.

https://doc.qt.io/qtcreator/creator-build-settings-qmake.html and https://doc.qt.io/qtcreator/creator-run-settings.html

**Reply by Mohammad-Imran01** [reply]
Thanks  aloooottt for your reply..

I have a QT application but Quick not Widget and want to integrate LNM inside it completely.
Use softkeys instead of buttons and mouse.

**Reply by Mohammad-Imran01** [reply]
![Screenshot (60)](https://github.com/albar965/littlenavmap/assets/117526105/39f6fd6e-4e5a-48cd-9001-37c60069e920)

**Reply by albar965** [reply]
Cool.  Integrating the whole program for map display would be a lot of effort. You could use the web api of LNM for this. https://github.com/KOKAProduktion/littlenavmap/wiki/Little-Navmap-Web-API-Documentation
Alex

**Reply by Mohammad-Imran01** [reply]
Sure, Thanks!!!! I need almost every function of the LNM thats why i am not willing to create whole of it from scratch........

**Reply by Mohammad-Imran01** [reply]
Hey can you help me again with this: 
![Screenshot (64)](https://github.com/albar965/littlenavmap/assets/117526105/27a6a2bc-5170-4d4e-a5f6-f6c4fbdd1af9)


If i manually paste libastro.dll and libmarblewidget-qt5.dll (because it says unable to find .....) and then open LNM.exe
![Screenshot (63)](https://github.com/albar965/littlenavmap/assets/117526105/51d0111d-f704-41e9-9f73-0f4c60332a0c)

**Reply by albar965** [reply]
Normally `make deploy` before running copies all needed stuff into `%APROJECTS%\deploy\...`. The libraries are put into the deploy LNM sub-folder. 

The second screen is shown since the folder `data` is missing in the installation folder. Again `make deploy` missing.

Make sure to run make `make deploy` and start the executable (you can also debug from there) from `%APROJECTS%\deploy\Little Navmap` (or similar, looking at Linux now).

**Reply by Mohammad-Imran01** [reply]
hey can you share build and run both settings screenshots..... 

![Screenshot (66)](https://github.com/albar965/littlenavmap/assets/117526105/f27bf72d-9528-4b89-ad35-7101e5c1fb96)

**Reply by Mohammad-Imran01** [reply]
1. click build
2. click run

---- right ??

**Reply by albar965** [reply]
These are from the VirtualBox VM I use to build LNM. Although the releases are built by scripts and not Qt Creator but I use it for debugging Windows problems.

The scripts are based on the APROJECTS environment variable and this is the only required var if all is put in the same paths.

Build settings for the 64-bit debug configuration:
![grafik](https://github.com/albar965/littlenavmap/assets/15254956/2aab340f-fbc3-41d6-9780-f788a3e8141c)

Run config:
![grafik](https://github.com/albar965/littlenavmap/assets/15254956/f676a21c-c4b5-4ac7-bf0c-4108f8bd4559)

Environment variables:
![grafik](https://github.com/albar965/littlenavmap/assets/15254956/5b932e2f-8f64-4f3b-9dd9-c3eaacf1653e)

Alex

**Comment by Mohammad-Imran01** [comment]
Hey, Thanks for your instructions, I am able to compile LNM very smoothly.
Now i can compile but,  it wont connect to the P3D simulator remote or same PC, can you tell me what am i missing for remote connection.
We are on same local network. 

The SS shows my variable locationg .lib files inside that is .dll files for simconnect
And on LNM its showing waiting for connect i am compiling LNM debug32

![Screenshot (1)](https://github.com/albar965/littlenavmap/assets/117526105/99d9151d-de38-4bb2-9978-7c448e0e96e9)

**Reply by Mohammad-Imran01** [reply]
I have copy pasted databases as mentioned in https://www.littlenavmap.org/manuals/littlenavmap/release/latest/en/NETWORK.html

**Reply by albar965** [reply]
The lib files and the path are only needed for compilation and deployment. LNM loads the SimConnect.dll dynamically (https://github.com/albar965/atools/blob/master/src/win/activationcontext.h) from the same folder as the executable. If the DLL is missing it will try to load one from the system.

This part of `make deploy` normally copies the DLL to the deployment folder:
https://github.com/albar965/littlenavmap/blob/47aaae37dfe6b1162f974c90d9a61400c9e4140d/littlenavmap.pro#L952-L958

You have to use the SimConnect.dll from FSX (https://www.littlenavmap.org/downloads/SimConnect/) since I do not support the 64-bit DLL for P3D yet.

Alex

**Reply by Mohammad-Imran01** [reply]
Thank you, 🫡🫡.. I will follow this one..

**Comment by Mohammad-Imran01** [comment]


**Reply by albar965** [reply]
Sorry, but I have no clue about Qt Quick. I don't use it. 

Have a look at the web api instead of adding this into LNM.

Alex

**Reply by Mohammad-Imran01** [reply]
Okay Thank you again,, Thanks alot for your work and support..

**Comment by Mohammad-Imran01** [comment]
Hey, i want to tell you that i am able to integrate my qml project into Littlenavmap using qquickwidget,.
Is there any simple step to remove things other than map even removing all menu  (form->gui->mainwindow.ui) If i remove them directly from Design section it gives alot of  error that is obvious.... I want the main window clean with map only.

**Reply by Mohammad-Imran01** [reply]
![Screenshot (4)](https://github.com/albar965/littlenavmap/assets/117526105/404afd0f-233a-44f8-9ea3-c1d34dec1b19)

**Reply by albar965** [reply]
Suggestion for a _fat_ approach keeping the whole program:
Maybe a layout file showing only the main window (https://www.littlenavmap.org/manuals/littlenavmap/release/latest/en/LAYOUT.html). This can hide all stuff like dock windows and toolbars but not the main menu. Main menu and context menu, tooltips and maybe more map interactions would need extra options to be disabled.
To hide toolbars and docks close all of them. Hide toolbars by right-clicking on them. The save a layout file which can be loaded on startup by command line (https://www.littlenavmap.org/manuals/littlenavmap/release/latest/en/COMMANDLINE.html).

This is the fastest approach. A _thin_ approach using only map display is a major reconstruction.

**Reply by Mohammad-Imran01** [reply]
Hey, Can you have a look. 
When i try to load qml files in qquickwidgets it works fine in any project other than LNM i am trying exactly same thing but it cannot find any qml libraries just in LNM maybe bcz deploy command and without deploy command it wont compile at all.

![Screenshot (6)](https://github.com/albar965/littlenavmap/assets/117526105/6c87c17d-8ffb-44b4-84af-cfe6b76a714b)
![Screenshot (5)](https://github.com/albar965/littlenavmap/assets/117526105/81e9f103-edcc-44e5-984f-5e1751860214)
![Screenshot (7)](https://github.com/albar965/littlenavmap/assets/117526105/49fc8470-6f23-43a2-aedd-b12915b93279)

**Reply by albar965** [reply]
You have to extend the deploy command to copy the QML libraries. LNM does not use QML per default.

**Reply by Mohammad-Imran01** [reply]
Hey can you please have a look, i have added the command to include quickcontrols2 and it is there .. but still same error
![Screenshot (8)](https://github.com/albar965/littlenavmap/assets/117526105/2036a54d-70ed-41fc-acfe-b3290dc5d2b8)

**Reply by Mohammad-Imran01** [reply]
![Screenshot (9)](https://github.com/albar965/littlenavmap/assets/117526105/ae518c57-8e99-4144-b7b3-8fd198460a1d)

**Reply by albar965** [reply]
Looks like the `libQt5QuickControls2.dll` and others are missing.
Maybe you even have to install these using the Qt Maintenance Tool and then include it into the project.
Alex

**Comment by Mohammad-Imran01** [comment]
Hey i have those qml , can you please tell me what to change in littlenavmap.pro, the command and where to write it.
It compiles fine but just cant fine qml libraries no matter what i do.

![Screenshot (16)](https://github.com/albar965/littlenavmap/assets/117526105/5db026ed-90b3-4870-9221-b7b5185f1b34)

**Reply by albar965** [reply]
Do no add the DLLs to the LIBS variable. This only for compiling. For runtime you have to add them to the installation folder (the one containing the littlenavmap.exe). Libraries are copied on "make deploy" right here: 
https://github.com/albar965/littlenavmap/blob/6e1e71cf45b376407041bd39cab5056ffe38af8d/littlenavmap.pro#L952-L976

BTW: You can also link into code directly as I did above. Just create a permalink from the code view in Github (select lines and click the tree dots) and add it here. Easier than screenshots and allows to copy and paste. :wink:

**Reply by Mohammad-Imran01** [reply]
https://github.com/Mohammad-Imran01/GNav1000/blame/7c8945bedb9447ed6c9e7e2dab9743a78316b50d/littlenavmap.pro#L985-L993

Hey as i told you no matter what i do it says qtquickcontrols not installed and wont load the qml libs.
![Screenshot (18)](https://github.com/albar965/littlenavmap/assets/117526105/522f52c6-d36b-409a-92c4-7472b347b965)
![Screenshot (17)](https://github.com/albar965/littlenavmap/assets/117526105/c20d13d5-f8b3-438f-a432-377c834abb39)

**Reply by Mohammad-Imran01** [reply]
Thanks for the sugesstion but i think i did something wrong... link only?
Please look at (line 81 in Lnm .pro) https://github.com/Mohammad-Imran01/GNav1000/blob/7c8945bedb9447ed6c9e7e2dab9743a78316b50d/littlenavmap.pro#L79-L84

**Reply by albar965** [reply]
I have no idea what is missing there. I'm afraid I cannot help much since I never use QML.

**Reply by Mohammad-Imran01** [reply]
Oh Okay, Thanks anyway!!! 👍👍👍👍, I'll try something else actually its working absolutely as expected if i go in debug folder and run.

**Comment by Mohammad-Imran01** [comment]
Thanks for your time and help.

**Reply by albar965** [reply]
You're welcome.
